### PR TITLE
Add support for Trino

### LINKED ISSUE #375: Add support for Trino
### Describe the feature
Add support for Trino, so this package could be used with [dbt-trino](https://github.com/starburstdata/dbt-trino) 

### Describe alternatives you've considered
Initially, we considered adding dbt-project-evaluator to [dbt-trino-utils package](https://github.com/starburstdata/dbt-trino-utils), which defines implementations of dispatched macros from other packages that can be used on Trino or Starburst.
However, we realised that it would be quite tedious to override all macros and models (and descendant models) on our side. It would also interfere with already existing packages in dbt-trino-utils. This approach would also require users to install dbt-trino-utils and make additional configurations in their dbt_project.yml file.

### Additional context
This is relevant for Trino.

### Who will this benefit?
Users who want to use dbt-project-evaluator with [dbt-trino](https://github.com/starburstdata/dbt-trino) 

### Are you interested in contributing this feature?
Yes!

### PR DESCRIPTION
**Pull request by damian3031**
This is a:
- [ ] bug fix PR with no breaking changes
- [x] new functionality

## Link to Issue 
Closes #375 

## Checklist
- [x] I have verified that these changes work locally on the following warehouses (Note: it's okay if you do not have access to all warehouses, this helps us understand what has been covered)
    - [ ] BigQuery
    - [ ] Postgres
    - [ ] Redshift
    - [ ] Snowflake
    - [ ] Databricks
    - [ ] DuckDB
    - [x] Trino
- [x] I have updated the README.md (if applicable)
- [ ] I have added tests & descriptions to my models (and macros if applicable)

### CODE REVIEW

**Comment by damian3031** [comment]
I've added integration tests to CircleCI. Please add below environment variables relevant for your Starburst Galaxy instance (AFAIK you have one) so that the tests can run on it.
`TRINO_TEST_HOST`, `TRINO_TEST_USER`, `TRINO_TEST_PASS`, `TRINO_TEST_PORT`, `TRINO_TEST_CATALOG_NAME`

**Comment by b-per** [comment]
Thanks @damian3031 ! 
I am testing it locally right now and we will add the Starburst credentials for CI/CD!

Based on my initial testing (using a Snowflake catalog) there are some issues (which might required updates to `dbt-trino`):
- if a table has a description, it is added to the DDL but I then get the error `TrinoUserError(type=USER_ERROR, name=NOT_SUPPORTED, message="This connector does not support creating tables with table comment", query_id=abcd)`
  - is it possible to disable the comment part of the DDL? it doesn't seem to be based on the code I saw in `dbt-trino`
- I am also getting some errors `This connector does not support creating views` which seems consistent with [this issue](https://github.com/trinodb/trino/issues/11588).

Based on the observations above:
- what catalog type would you recommend to avoid those errors? (or which one did you test with for now)
- should we test the package for different catalog types?
- we might need to add some info in the README to mention that some catalog types will work with Trino but not all

**Comment by damian3031** [comment]
@b-per creating views is supported in object storage connectors such as Hive, Delta Lake, and Apache Iceberg.

I tested it on the Iceberg connector, and the tests passed.

I think it would be fair to test it specifically against Iceberg connector and mention it in the README.

File: README.md [hunk 1]
```diff
@@ -20,6 +20,7 @@ Currently, the following adapters are supported:
 - Redshift
 - Snowflake
 - DuckDB
+- Trino
```
Comment (damian3031) [inline README.md:23]:
```suggestion
- Trino (tested with Iceberg connector)
```
@b-per Would that be ok?

File: .circleci/config.yml [hunk 2]
```diff
@@ -104,3 +115,4 @@ workflows:
           context: profile-databricks
           requires:
             - integration-postgres
+      - integration-trino
```
Comment (b-per) [inline .circleci/config.yml:118]: Could you please just add a 
```
          requires:
            - integration-postgres
```
like the other Cloud Warehouses? The idea is that we try first with the "offline" ones so we don't incur any Cloud cost if those fail.

**Review by b-per — CHANGES_REQUESTED** [review]
Thanks for the work here!

All good with me with the small exception of updating the CI config file to run Trino at the same time as Snowflake/BQ etc...

I have added the env vars to test on S3 and it works

File: .circleci/config.yml (hunk 2 above)
Comment (damian3031) [reply .circleci/config.yml]: Done!
I've also added `context: profile-trino`

**Review by b-per — APPROVED** [review]

**Comment by b-per** [comment]
We'll do a new release of the package on the dbt Hub in just a beat so that it is accessible to users.

**Pull request merged**